tamr_toolbox/data_io/cloud.py

```diff
@@ -0,0 +1,220 @@
+import tempfile
+import tarfile
+
+
+def gcs_upload(
+    cloud_client,
+    source_filepath="path_to_my_local_file",
+    destination_filepath="path_to_my_file_in_bucket",
+    bucket_name="sample_bucket",
+    tar_file=True,
+    return_uploaded_file=None,
+):
+    """Upload data to a Google storage bucket
+    Args:
+        source_filepath: file path of source file being uploaded
+        destination_filepath: path to google bucket
+        cloud_client: google storage client with user credentials
+        bucket_name: name of google bucket
+        tar_file: Tar file before upload
+        return_uploaded_file: If filepath is given,
+        downloads copy of file being uploaded to filepath
+
+
+    """
+    bucket = cloud_client.get_bucket(bucket_name)
+    blob = bucket.blob(destination_filepath)
+
+    if tar_file:
+        with tempfile.NamedTemporaryFile(delete=False, mode="w") as tmp:
+            with tarfile.open(tmp.name, "w") as tar:
+                tar.add(source_filepath)
+            blob.upload_from_filename(tmp.name)
+            if return_uploaded_file:
+                with tarfile.open(return_uploaded_file, "w") as tar:
+                    tar.add(source_filepath)
+
+    else:
+        blob.upload_from_filename(source_filepath)
+        if return_uploaded_file:
+            with tarfile.open(return_uploaded_file, "w") as tar:
+                tar.add(source_filepath)
```

Comment:
If I understand correctly, the `tar_file` functionality is just for tarring up a file before uploading it (and for download functions is just about untarring a downloaded file).

This is cool functionality that is probably useful, since it allows uploading a collection of files, but in this implementation it over-complicated the code.

Can you simplify the upload and download functions so that they *only* upload/download a single file?  The tar/untar should be its own function, which could be used in combination with the simpler upload/download to have the same effect with more readability.